tools/language-server/logstat/src/main.rs

```diff
@@ -0,0 +1,322 @@
+// === Standard Linter Configuration ===
+#![deny(non_ascii_idents)]
+#![warn(unsafe_code)]
+
+use clap::Parser;
+use clap::ValueHint;
+use lazy_static::lazy_static;
+use regex::Regex;
+use std::fmt;
+use std::io::Result;
+use std::path::PathBuf;
+use std::process;
+use time::format_description::well_known::Rfc3339;
+use time::Duration;
+use time::OffsetDateTime;
+use tokio::fs::File;
+use tokio::io::AsyncBufReadExt;
+use tokio::io::BufReader;
+use tokio_stream::wrappers::LinesStream;
+use tokio_stream::StreamExt;
+
+
+
+#[derive(Parser, Debug)]
+#[clap(version, about)]
+struct Args {
+    /// Logfile to analyze
+    #[clap(value_name = "FILE", value_hint = ValueHint::FilePath)]
+    log: PathBuf,
+
+    /// Specification file
+    #[clap(long, value_name = "FILE", value_hint = ValueHint::FilePath)]
+    spec: PathBuf,
+
+    /// Wstest log file
+    #[clap(long, value_name = "FILE", value_hint = ValueHint::FilePath)]
+    wstest_log: Option<PathBuf>,
+
+    /// Number of iterations to skip
+    #[clap(long, default_value = "0")]
+    skip_iterations: usize,
+
+    /// Calculate median instead of mean
+    #[clap(long)]
+    median: bool,
+}
+
+/// A specification containing lines to lookup in the log file
+#[derive(Debug)]
+struct Spec {
+    matches: Vec<String>,
+}
+
+/// Timed operation
+#[derive(Debug)]
+struct Operation {
+    duration:  Duration,
+    timestamp: OffsetDateTime,
+    line:      String,
+}
+
+impl fmt::Display for Operation {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let duration_millis = self.duration.whole_milliseconds();
+        let timestamp = self.timestamp.format(&Rfc3339).unwrap();
+        let truncated_line = self.line.chars().take(80).collect::<String>();
+
+        write!(f, "{}ms [{}] {}...", duration_millis, timestamp, truncated_line)
+    }
+}
+
+/// Single iteration containing multiple sub-operations
+#[derive(Debug)]
+struct Iteration {
+    pub operations: Vec<Operation>,
+}
+
+impl Iteration {
+    pub fn total_time(&self) -> Duration {
+        let mut total: Duration = Duration::ZERO;
+        for operation in &self.operations {
+            total += operation.duration
+        }
+        total
+    }
+}
+
+/// Final statistics about benchmarked operation
+#[derive(Debug)]
+struct Stats {
+    min:  Duration,
+    max:  Duration,
+    avg:  Duration,
+    line: String,
+}
+
+impl fmt::Display for Stats {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let avg_millis = self.avg.whole_milliseconds();
+        let min_millis = self.min.whole_milliseconds();
+        let max_millis = self.max.whole_milliseconds();
+        let truncated_line = self.line.chars().take(80).collect::<String>();
+
+        write!(
+            f,
+            "{avg}ms [{min}..{max}] {line}",
+            avg = avg_millis,
+            min = min_millis,
+            max = max_millis,
+            line = truncated_line
+        )
```

Comment:
You can inline variable names in string, like this:
```suggestion
        write!(f, "{avg_millis}ms [{min_millis}..{max_millis}] {truncated_line}")
```